### DISCUSSION TITLE
Increase volume boost 2-3x. Add max volume limit/cutoff.

### DISCUSSION THREAD

**Discussion by mars4science** [PaulWoitaschek/Voice#1712] [Ideas]
Now volume boost is written to be up to 9 db. It is noticeable, but no much. Recently I rarely been in noisy environments and the files I listen to have "large" volume/range to start with. But for the future, as boost is already in the menu, IMO it makes sense to increase range 2-3x from current one. 

Another more complex thing is adding cutoff (max volume), which is not straight forward AFAIK. My "problem" some files have short very loud sci-fi like music inclusions which I'd like to be able to limit to much lower volume. How difficult can it be?  TIA

**Comment by decadent** [comment]
Volume boost is a feature of a built-in Android audio player used from apps, and boost is limited to 9 dB (to my knowledge). There's basically no chance that Mr. Woitaschek makes his own audio player or audio processing, as it involves a lot of low-level code with a ton of edge cases and decisions that need to be accounted for—and which are already handled by the standard player. Thus, I wouldn't count on this ever being implemented.

For raising volume of specific books, I just process the files on the desktop with `mp3gain`—there are also `vorbisgain` and `aacgain`, these all raise volume without recompressing the audio. There are probably GUI tools, at least frontends for `mp3gain`.

As an alternative, there are Android apps that process all system audio, e.g. for equalizing, and there are several of them for boosting volume—they're handy since multiple various audio-playing apps are in the same position as Voice with the lack of effects. I'm using the Wavelet equalizer app, because basic features are free and without unnecessary permissions—and it seems that it has a ‘limiter’, which might work both as a dynamic compressor (what you're calling cutoff), and to just raise the volume with post-gain, if the compression ratio is set to 1. Granted, this approach requires enabling the effect when listening to an audiobook, and disabling it after.

Dynamic range compression is also possible on the desktop, of course, though it would necessitate recompressing the files. I would use command-line `ffmpeg` or `sox` for this, but again there are probably GUI tools that can do batch-processing.

**Comment by mars4science** [comment]
@decadent , thanks. Do you know if speed change is also Android feature? I'd like more range for speed. Is 0.5 slowest?
